engine/consensus/matching/core.go

```diff
@@ -541,19 +539,18 @@ func (c *Core) CheckSealing() error {
 		return fmt.Errorf("could not request pending result approvals: %w", err)
 	}
 
-	mempoolHasNextSeal := false
-
 	// check if we end up created a seal in mempool for the next unsealed block
-	for _, unsealed := range nextUnsealeds {
-		_, mempoolHasNextSeal = c.seals.ByID(unsealed.IncorporatedResultID)
+	mempoolHasNextSeal := false
+	for _, nextUnsealed := range sealingTracker.Records() {
+		_, mempoolHasNextSeal = c.seals.ByID(nextUnsealed.IncorporatedResult.ID())
 		if mempoolHasNextSeal {
 			break
 		}
 	}
-
-	lg.Info().
+	c.log.Info().
+		Int("sealable_results_count", len(sealableResults)).
 		Int("sealable_incorporated_results", len(sealedBlockIDs)).
-		Int64("duration_ms", time.Since(startTime).Milliseconds()).
+		Str("next_unsealed_results", sealingTracker.String()).
```

Comment:
Does `sealingTracker.String()` include all records, or just the next unsealed result?

Could you provide a sample log?

Better not all records, because there will be too much noise, as we only care about the next unsealed when debugging.

Reply:
we only record the sealing status of the next unsealed finalized block.

Reply:
The logging behaviour should be almost identical to the initial version you built: https://github.com/onflow/flow-go/pull/363#discussion_r569732038


## When missing approval for the next unsealed block
The log shows we have the result, but the FirstUnmatchedChunkIndex is `0`. Meaning, we didn't receive approval for it.

```
[{
"executed_block_height":2190957487,
"executed_block_id":"9efa9526c0ab51db9c5a904bfc589f95955c1659f6b439196c0146dc8e2a3694",
"first_unmatched_chunk_index":2,
"incorporated_result_id":"b159f2623bd9d660fa1125d05487d7658e5725fcba479568da139e880df8422e",
"number_chunks":17,
"qualifies_for_emergency_sealing":false,
"result_id":"53c8d0dd81bf4aef407331c2388bcdd53f4bd5ae5cb81ad5242a9ee6655706db",
"sufficient_approvals_for_sealing":false}]
```

## When sealed by emergency
The `SealedByEmergency` becomes true

```
[{
"executed_block_height":2190957487,
"executed_block_id":"9efa9526c0ab51db9c5a904bfc589f95955c1659f6b439196c0146dc8e2a3694",
"first_unmatched_chunk_index":2,
"has_multiple_receipts":true,
"incorporated_result_id":"b159f2623bd9d660fa1125d05487d7658e5725fcba479568da139e880df8422e",
"number_chunks":17,
"qualifies_for_emergency_sealing":true,
"result_id":"53c8d0dd81bf4aef407331c2388bcdd53f4bd5ae5cb81ad5242a9ee6655706db",
"sufficient_approvals_for_sealing":false}]
```

Reply:
Looks good. Thanks for providing the log